File: backend/src/main/java/com/pethouse/review/service/ReviewService.java
package com.pethouse.review.service;

import com.pethouse.booking_hai.dto.response.BookingResponse;
import com.pethouse.review.dto.request.ReviewRequest;
import com.pethouse.review.dto.response.ReviewResponse;

import java.util.List;

public interface ReviewService {

    // Lấy danh sách reviews công khai (cho HomePage)
    List<ReviewResponse> getPublicReviews();

    // Lấy reviews của user hiện tại
    List<ReviewResponse> getMyReviews();

    // Lấy danh sách booking đủ điều kiện đánh giá (COMPLETED + chưa có review)
    List<BookingResponse> getEligibleBookings();

    // Tạo review mới
    ReviewResponse createReview(ReviewRequest request);

    // Cập nhật review
    ReviewResponse updateReview(Long reviewId, ReviewRequest request);

    // Xóa review
    void deleteReview(Long reviewId);

    // Lấy review theo ID
    ReviewResponse getReviewById(Long reviewId);
}

File: frontend/src/features/review_huyen/services/reviewService.js
import httpClient from '../../../api/httpClient';

const reviewService = {
    // Lấy reviews công khai (cho HomePage)
    getPublicReviews: async () => {
        const response = await httpClient.get('/reviews');
        return response.data;
    },

    // Lấy reviews của user hiện tại
    getMyReviews: async () => {
        const response = await httpClient.get('/reviews/my');
        return response.data;
    },

    // Lấy danh sách booking đủ điều kiện đánh giá
    getEligibleBookings: async () => {
        const response = await httpClient.get('/reviews/eligible-bookings');
        return response.data;
    },

    // Tạo review mới
    createReview: async (reviewData) => {
        const response = await httpClient.post('/reviews', reviewData);
        return response.data;
    },

    // Cập nhật review
    updateReview: async (reviewId, reviewData) => {
        const response = await httpClient.put(`/reviews/${reviewId}`, reviewData);
        return response.data;
    },

    // Xóa review
    deleteReview: async (reviewId) => {
        const response = await httpClient.delete(`/reviews/${reviewId}`);
        return response.data;
    },

    // Lấy review theo ID
    getReviewById: async (reviewId) => {
        const response = await httpClient.get(`/reviews/${reviewId}`);
        return response.data;
    }
};

export default reviewService;
